frontend/js/ci.js

```diff
@@ -136,9 +219,79 @@ const displayGraph = (runs) => {
     window.onresize = function () { // set callback when ever the user changes the viewport
         chart_instance.resize();
     }
+
+    return chart_instance;
 }
 
+const displayAveragesTable = (runs) => {
+    let labels = new Set()
+    runs.forEach(run => {
+        let [value, unit, run_id, timestamp, label, cpu, commit_hash, duration] = run;
+        labels.add(label)
+    });
+
+    const tableBody = document.querySelector("#label-avg-table");
+    tableBody.innerHTML = "";
+
+    const label_total_avg_node = document.createElement("tr")
+    label_total_avg_node.innerHTML += `
+                            <td class="td-index">${getTotalAverage(runs)} mJ</td>
+                            <td class="td-index">Total</td>`
+    tableBody.appendChild(label_total_avg_node);
+
+    labels.forEach(label => {
+        const label_avgs_node = document.createElement("tr")
+        let avg = getAverageOfLabel(runs, label);
+        label_avgs_node.innerHTML += `
+                                        <td class="td-index">${avg} mJ</td>
+                                        <td class="td-index">${label}</td>`
+    document.querySelector("#label-avg-table").appendChild(label_avgs_node);
+    });
+}
+
+const displayCITable = (runs, url_params) => {
+    runs.forEach(el => {
+        const li_node = document.createElement("tr");
+
+        [badge_value, badge_unit] = convertValue(el[0], el[1])
+        const value = badge_value + ' ' + badge_unit;
+
+        const run_id = el[2];
+        const cpu = el[5];
+        const commit_hash = el[6];
+        const short_hash = commit_hash.substring(0, 7);
+        const tooltip = commit_hash.length > 7 ? `title="${commit_hash}"` : '';
```

Comment:
Why would the commit hash ever be shorter than 7 characters?
And why is it not just always the title?

Reply:
Hmm, good call. I simplified this to just title="${commit_hash}.